pkg/routeagent/controllers/route/route.go

```diff
@@ -173,17 +173,20 @@ func (r *Controller) Run(stopCh <-chan struct{}) error {
 		return fmt.Errorf("failed to wait for caches to sync")
 	}
 
-	r.cniIface = discoverCNIInterface(r.localClusterCidr[0])
-	if r.cniIface != nil {
+	cniIface, err := discoverCNIInterface(r.localClusterCidr[0])
+	if err != nil {
```

Comment:
Think you reversed the check. It should be if err == nil or if r.cniIface != nil.

Reply:
right